### ISSUE TITLE
Core dumps

### ISSUE THREAD

**Issue by st1618** [skilion/onedrive#138]
First things first, thank you for taking the time to create and maintain this code, works like a charm most of the time...

Been having problems when running in monitor mode when deleting files, started to get core dumps from the process cause by an item not found exception being thrown in itemdb.d, i've looked at this and i seems to be being caused by MS returning items that have been marked for deletion but the code is trying to make a path for it from the database...

Anyway have hacked about it the code and changing this (line 162 - sync.d)

```
		string path = ".";
		if (parentId) {
			path = itemdb.computePath(parentId) ~ "/" ~ name;
		}
```

to

```
		string path = ".";
		if (parentId && !isItemDeleted(item)) {
			path = itemdb.computePath(parentId) ~ "/" ~ name;
		}
```

seems to resolve the problem for me.

Kind Regards

**Comment by joecool1029** [comment]
This was a big help for me too!

Do note you will have to run a onedrive --resync after rebuilding the new binary with this option.

**Comment by hugomilan** [comment]
I applied this change and then did 'onedrive' until I got all my files synchronized. Then, I did 'onedrive --resync' and finally did 'onedrive -m'. It is working now, although I saw some error messages.

I observed the following when running 'onedrive --resync', although it did not interfere with the synchronization.
 
HTTP request returned status code 502 (Bad Gateway)
null


When running 'onedrive -m', although it did not seem to interfere with the synchronization either, I observed:
HTTP request returned status code 400 (Bad Request)
{
    "error": {
        "code": "invalidRequest",
        "message": "Bad Argument"
    }
}

**Comment by joecool1029** [comment]
Unfortunately it still crashes frequently as you've noticed. But for other reasons. 

Here's a bash script that restarts it when it dies: 

```
until onedrive -m; do
    echo "Server 'onedrive' crashed with exit code $?.  Respawning.." >&2
    sleep 1
done

**Comment by hugomilan** [comment]
Thanks for the bash script.

When I see the "status code 400", onedrive keeps running. It doesn't crash.

**Comment by skilion** [comment]
Fixed in 957b034805985d40a25d8521996791c74c055a64
